carsus/io/cmfgen/cmfgen_.py

```diff
@@ -0,0 +1,199 @@
+    """
+        Description
+        ----------
+        base : pandas.DataFrame
+        columns : list of str
+            (default value = COLUMNS)
+            
+        Methods
+        -------
+        load(fname)
+            Parses the input data and stores the results in the `base` attribute
+    """
+
+    def load(self, fname):
+        args = {}
+        args['header'] = None
+        args['index_col'] = False
+        args['sep'] = '\s*\|\s*|-?\s+-?|(?<=[^ED\s])-(?=[^\s])'
+        args['skiprows'] = find_row(fname, "Transition", "Lam", num_row=True) +1
+    
+        n = find_row(fname, "Number of transitions").split()[0]
+        args['nrows'] = int(n)
+
+        columns = ['State A', 'State B', 'f', 'A', 'Lam(A)', 'i', 'j', 'Lam(obs)', '% Acc']
+    
+        try:
+            df = pd.read_csv(fname, **args, engine='python')
+    
+        except pd.errors.EmptyDataError:
+            df = pd.DataFrame(columns=columns)
+            warnings.warn('Empty table')
+    
+        # Assign column names by file content
+        if df.shape[1] == 9:
+            df.columns = columns
+        
+        elif df.shape[1] == 10:
+            df.columns = columns + ['?']
+            df = df.drop(columns=['?'])
+    
+        elif df.shape[1] == 8:
+            df.columns = columns[:-2] + ['#']
+            df = df.drop(columns=['#'])
+            df['Lam(obs)'] = np.nan
+            df['% Acc'] = np.nan
+    
+        else:
+            warnings.warn('Inconsistent number of columns')       
+    
+        # Fix for Fortran float type 'D'
+        if df.shape[0] > 0 and 'D' in str(df['f'][0]):
+            df['f'] = df['f'].map(to_float)
+            df['A'] = df['A'].map(to_float)
+
+        self.base = df
+        self.columns = df.columns.tolist()
+
+
+class CMFGENCollitionalDataParser(BaseParser):
+    """
+        Description
+        ----------
+        base : pandas.DataFrame
+        columns : list of str
+            (default value = COLUMNS)
+            
+        Methods
+        -------
+        load(fname)
+            Parses the input data and stores the results in the `base` attribute
+    """
+
+    def load(self, fname):
+    
+        args = {}
+        args['header'] = None
+        args['index_col'] = False
+        args['sep'] = '\s*\|\s*|-?\s+-?|(?<=[^ED\s])-(?=[^\s])'  # FIXME: This is the same regex used in OSC files, change it
+        args['skiprows'] = find_row(fname, "ransition\T", num_row=True)
+
+        try:
+            n = find_row(fname, "Number of transitions").split()[0]
+            args['nrows'] = int(n)
+    
+        except AttributeError:
+            pass
+    
+        try:
+            names = find_row(fname, 'ransition\T').split()
+            names = [ n.replace('D', 'E') for n in names ]
+            args['names'] = ['State A', 'State B'] + names[1:]
+
+        except AttributeError:
+            warnings.warn('No column names')  # Some files have no column names nor header
```

Comment:
I think at some point we want to assign column names here. Consider adding a TODO.